internal/engine/wazevo/frontend/lower.go

```diff
@@ -475,18 +470,49 @@ func (c *Compiler) lowerOpcode(op wasm.Opcode) {
 		wasm.OpcodeI64TruncF64U, wasm.OpcodeI64TruncF32U,
 		wasm.OpcodeI32TruncF64U, wasm.OpcodeI32TruncF32U:
 		if state.unreachable {
-			return
+			break
 		}
 		ret := builder.AllocateInstruction().AsFcvtToInt(
 			state.pop(),
 			c.execCtxPtrValue,
 			op == wasm.OpcodeI64TruncF64S || op == wasm.OpcodeI64TruncF32S || op == wasm.OpcodeI32TruncF32S || op == wasm.OpcodeI32TruncF64S,
 			op == wasm.OpcodeI64TruncF64S || op == wasm.OpcodeI64TruncF32S || op == wasm.OpcodeI64TruncF64U || op == wasm.OpcodeI64TruncF32U,
+			false,
 		).Insert(builder).Return()
 		state.push(ret)
+	case wasm.OpcodeMiscPrefix:
+		c.loweringState.pc++
+		// A misc opcode is encoded as an unsigned variable 32-bit integer.
+		miscOpUint, num, err := leb128.LoadUint32(c.wasmFunctionBody[c.loweringState.pc:])
+		if err != nil {
+			// fixme return err
+			panic(fmt.Sprintf("failed to read misc opcode: %v", err))
+		}
```

Comment:
in current compiler the procedure returns an error, should we do it here too? @mathetake

Reply:
no, this comes after function validation, so panic should be fine and shouldn't happen. See the code pass of wazero.Runtime.CompileModule!